x/concentrated-liquidity/swaps.go

```diff
@@ -134,6 +182,30 @@ func (k Keeper) SwapOutAmtGivenIn(ctx sdk.Context, tokenIn sdk.Coin, tokenOutDen
 	return tokenOut, nil
 }
 
+func (k *Keeper) SwapInAmtGivenOut(
+	ctx sdk.Context,
+	desiredTokenOut sdk.Coin,
+	tokenInDenom string,
+	swapFee sdk.Dec,
+	priceLimit sdk.Dec,
+	poolId uint64,
+) (tokenIn sdk.Coin, err error) {
+	tokenIn, actualTokenOut, updatedTick, updatedLiquidity, updatedSqrtPrice, err := k.calcInAmtGivenOut(ctx, desiredTokenOut, tokenInDenom, swapFee, priceLimit, poolId)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	if err := k.applySwap(ctx, tokenIn, actualTokenOut, poolId, updatedLiquidity, updatedTick, updatedSqrtPrice); err != nil {
+		return sdk.Coin{}, err
+	}
+
+	if tokenIn.Amount.GT(tokenIn.Amount) {
```

Comment:
This is checking against the same value. Do we have tests covering this logic?

Reply:
This is an issue from a rebase that I missed, removed